developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -385,7 +388,7 @@ export class KmnCompiler implements UnicodeSetParser {
     let writer = new KvkFileWriter();
     return {
       filename: this.callbacks.path.join(this.callbacks.path.dirname(kmxFilename),
-        this.callbacks.path.basename(kvksFilename, KeymanFileTypes.Source.VisualKeyboard) + KeymanFileTypes.Binary.VisualKeyboard),
+        basename + KeymanFileTypes.Binary.VisualKeyboard),
```

Comment:
Uh... isn't this going to result in a filename with two extensions?  As far as I can see, `basename` gets assigned either a `.kvks` or a `.kvk` filename, assuming I'm properly inferring what `this.callbacks.path.basename` does.

To clarify what I'm inferring:

```javascript
this.callbacs.path.basename('my_keyboard.kmn', KeymanFileTypes.Binary.VisualKeyboard')
```

would return `my_keyboard.kvk` - stripping off the base file-extension (`kmn`) and replacing it with the constant `KeymanFileTypes.Binary.VisualKeyboard`, which resolves to `.kvk`.

In which case... at line 391, as there is no `.basename` call, we could get `my_keyboard.kvk.kvk`.  Unless I'm missing something?

Reply:
`paths.basename` strips off an extension if it matches the provided extension. It's ugly, but that's what it does, so:

```js
const KeymanFileTypes.Binary.VisualKeyboard = '.kvk';
this.callbacks.path.basename('my_keyboard.kvk', KeymanFileTypes.Binary.VisualKeyboard) == 'my_keyboard'
```